services/venice-router/src/main/java/com/linkedin/venice/router/api/path/VeniceComputePath.java

```diff
@@ -150,6 +150,29 @@ public final RequestType getStreamingRequestType() {
     return RequestType.COMPUTE_STREAMING;
   }
 
+  public VeniceMultiGetPath toMultiGetPath() {
```

Comment:
Cost wise this is similar to sub-path creation during normal Router operation, i.e not very expensive. Also, this code path is only active during short periods of time when client and service is out of sync regarding the state of Read-Compute. This I expect the overall overhead to be low.

Regarding the possible optimization we can deprecate `ComputeRouterRequestKeyV1` and use `MultiGetRouterRequestKeyV1` instead. It'd even be a backward compatible change since both records are binary compatible. The downside is that such change would tie multi-get and compute transport protocols.

Reply:
I'm open to benchmark it with JMH and implement this optimization in a separate PR. @gaojieliu what do you think about it?

Reply:
As @anaberezhnov mentioned, this function won't be executed in the hot path, so I don't have a strong preference.